modules/plugin/geopkg/src/main/java/org/geotools/geopkg/GeoPkgDialect.java

```diff
@@ -995,6 +1020,21 @@ public Object convertValue(Object value, AttributeDescriptor ad) {
                     ad.getType().getBinding().getComponentType(),
                     (EnumMapper) ad.getUserData().get(GPKG_ARRAY_ENUM_MAP));
         }
+        if (ad.getType().getBinding() == Timestamp.class) {
+            if (value == null) {
+                return null;
+            }
+            // geopkg - format will be ISO-8601 - YYYY-MM-DDTHH:MM[:SS.SSS]Z
+            var strValue = (String) value;
+            // this is for support with "bad" geopkgs
+            if (!strValue.endsWith("Z")) {
+                strValue += "Z";
+            }
+            Instant instant = Instant.parse(strValue);
+            Timestamp timestamp = Timestamp.from(instant);
+            return timestamp; // this will be in local time
+        }
```

Comment:
Makes sense, but isn't this overly rigid? I would not read anymore geopackages written by a previous version of GeoTools, or files written without strict adherence to the the GeoPackage standard. I'd keep the code within try/catch, and fall back on Converters (what ``super.convertValue`` is doing already) to try and interpret values in a less strict version.

Generally speaking, it's what we try to do in GeoTools data reading in general, be lenient and try to read the input, rather than playing "specification cop".